lib/cli-engine.js

```diff
@@ -744,12 +874,17 @@ class CLIEngine {
      * This is the same logic used by the ESLint CLI executable to determine
      * configuration for each file it processes.
      * @param {string} filePath The path of the file to retrieve a config object for.
-     * @returns {Object} A configuration object for the file.
+     * @returns {ConfigData} A configuration object for the file.
      */
     getConfigForFile(filePath) {
-        const configHelper = this.config;
-
-        return configHelper.getConfig(filePath);
+        const { configArrayFactory, options } = internalSlotsMap.get(this);
+        const absolutePath =
+            filePath ? path.resolve(options.cwd, filePath) : options.cwd;
```

Comment:
This is for backward compatibility. Several tests have called `getConfigForFile` method without arguments...

Reply:
I think it would be reasonable to require the argument since this is a breaking change anyway and calling it with no arguments is undocumented. But I don't feel strongly about it.

Reply:
This was my memory's confusion; tests without arguments were for old `configHelper.getConfig(filePath)`. I have removed this check.